tests/backends/data/test_es.py

```diff
@@ -0,0 +1,638 @@
+        LOCALE_ENCODING="utf-16",
+    )
+    backend = ESDataBackend(settings)
+    assert backend.default_index == ES_TEST_INDEX
+    assert backend.locale_encoding == "utf-16"
+    assert isinstance(backend.client, Elasticsearch)
+    elasticsearch_node = backend.client.transport.node_pool.get()
+    assert elasticsearch_node.config.ca_certs == Path("/path/to/ca/bundle")
+    assert elasticsearch_node.config.verify_certs is True
+    assert elasticsearch_node.host == "elasticsearch_hostname"
+    assert elasticsearch_node.port == 9200
+
+    try:
+        ESDataBackend(settings)
+    except Exception as err:  # pylint:disable=broad-except
+        pytest.fail(f"ESDataBackend should not raise exceptions: {err}")
+
+
+def test_backends_data_es_data_backend_status_method(monkeypatch, es_backend, caplog):
+    """Test the `ESDataBackend.status` method."""
+    backend = es_backend()
+    with monkeypatch.context() as elasticsearch_patch:
+        # Given green status, the `status` method should return `DataBackendStatus.OK`.
+        es_status = "1664532320 10:05:20 docker-cluster green 1 1 2 2 0 0 1 0 - 66.7%"
+        elasticsearch_patch.setattr(backend.client, "info", lambda: None)
+        elasticsearch_patch.setattr(backend.client.cat, "health", lambda: es_status)
+        assert backend.status() == DataBackendStatus.OK
+
+    with monkeypatch.context() as elasticsearch_patch:
+        # Given yellow status, the `status` method should return
+        # `DataBackendStatus.ERROR`.
+        es_status = "1664532320 10:05:20 docker-cluster yellow 1 1 2 2 0 0 1 0 - 66.7%"
+        elasticsearch_patch.setattr(backend.client, "info", lambda: None)
+        elasticsearch_patch.setattr(backend.client.cat, "health", lambda: es_status)
+        assert backend.status() == DataBackendStatus.ERROR
+        # Given yellow status, and `settings.ALLOW_YELLOW_STATUS` set to `True`,
+        # the `status` method should return `DataBackendStatus.OK`.
+        backend.settings.ALLOW_YELLOW_STATUS = True
+        with caplog.at_level(logging.INFO):
+            assert backend.status() == DataBackendStatus.OK
+
+        assert (
+            "ralph.backends.data.es",
+            logging.INFO,
+            "Cluster status is yellow.",
+        ) in caplog.record_tuples
+
+    # Given a connection exception, the `status` method should return
+    # `DataBackendStatus.ERROR`.
+    with monkeypatch.context() as elasticsearch_patch:
+
+        def mock_connection_error():
+            """ES client info mock that raises a connection error."""
+            raise ESConnectionError("Mocked connection error")
+
+        elasticsearch_patch.setattr(backend.client, "info", mock_connection_error)
+        with caplog.at_level(logging.ERROR):
+            assert backend.status() == DataBackendStatus.AWAY
+
+    assert (
+        "ralph.backends.data.es",
+        logging.ERROR,
+        "Failed to connect to Elasticsearch: Connection error",
```

Comment:
Why is the message used in the mocked method not returned? It should be the following message, no ? 
```
Failed to connect to Elasticsearch: Mocked connection error
```

Reply:
Good point) This was due to the way the `ESConnectionError` was initialized (missing second argument), updated the test to make the error message less confusing)